pkg/backend/httpstate/backend.go

```diff
@@ -1234,6 +1260,51 @@ func (b *cloudBackend) PromptAI(
 	return res, nil
 }
 
+func (b *cloudBackend) ShowCopilotErrorSummary(
+	ctx context.Context, pulumiOutput []string, stackRef backend.StackReference, opts display.Options,
+) {
+	summary, err := b.DoSummarizeErrorWithCopilot(ctx, pulumiOutput, stackRef, opts)
+	fmt.Println(opts.Color.Colorize(colors.SpecHeadline + "Summary:" + colors.Reset))
+	if err != nil {
+		fmt.Printf("error summarizing update output: %s", err)
+	}
+	for _, line := range summary {
+		fmt.Println("  " + opts.Color.Colorize(colors.BrightGreen+line+colors.Reset))
+	}
+	// nice empty line
+	fmt.Println()
+}
+
+func (b *cloudBackend) DoSummarizeErrorWithCopilot(
```

Comment:
Does this need to be public? Worth a doc comment on this/the function above?

Reply:
Equally, not sure if this could be inlined into the function above/what the value is in pulling it out

Reply:
I find myself doing this a lot with "the fn that can handle and return lots of errors" + "the fn that calls that fn and handles them all in one spot".

Which makes for a funny logic split sometimes.

Will try and clean it up a bit though..